### PR TITLE
Add OP Plasma to devnet

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This PR adds the DA server to the devnet. The op-node and op-batcher are also configured to use the plasma feature when enabled via the env var `DEVNET_PLASMA=true`. There is a stub for sentinel which works locally, but there is no public image for Sentinel yet. The da-server is also now built as part of the op-stack-go build process.

In addition, I had to disable the devnet with fault proofs because the matrix parameter was wrong and the FP devnet was not working when I actually enabled them via the parameter (they are timing out).

Last change: This also re-enables OP Plasma e2e test which got disabled in https://github.com/ethereum-optimism/optimism/pull/10058

**Tests**

Tested locally.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1772,18 +1795,27 @@ workflows:
...
+          save_image_tag: <<pipeline.git.revision>> # for devnet later
       - cannon-prestate:
           requires:
             - go-mod-download
       - devnet:
+          name: devnet<< matrix.fpac >>
           matrix:
             parameters:
-              fpac: ["legacy", "fault-proofs"]
+              # TODO(10142): We need to re-enable "-fault-proofs" in the future
```
Comment (ajsutton) [inline .circleci/config.yml]: Fixed in https://github.com/ethereum-optimism/optimism/pull/10144

File: .circleci/config.yml [hunk 2]
```diff
@@ -1772,18 +1795,27 @@ workflows:
...
+              fpac: ["-legacy", "-plasma"]
```
Comment (ajsutton) [inline .circleci/config.yml]: Personally, I'd omit the `-` prefix on all of these. Then you don't need to override the `name` as CircleCI will concat `{name}-{params}` for you (which is how we get `devnet-fault-proofs` currently.  Just update the other side to not have the `-` prefix either. Doesn't really matter either way though.

File: ops-bedrock/docker-compose.yml [hunk 3]
```diff
@@ -197,6 +202,36 @@ services:
...
+    ports:
+      - "3100:3100"
+    volumes:
+      - "l2_data:/data"
+
+  sentinel:
+    image: quarry/sentinel # TODO(10141): We need a public image for this (sentinel is out of repo)
+    volumes:
+      - "${PWD}/../.devnet/rollup.json:/rollup.json"
+    # private-key is mnemonic index 5 of the test mnemonic.
```
Comment (ajsutton) [inline ops-bedrock/docker-compose.yml:227]: Are we able to pass the mnemonic like we do for `op-challenger`? https://github.com/ethereum-optimism/optimism/blob/f2c6debe7a575d49deaef0b1635bf3f4dc70901b/ops-bedrock/docker-compose.yml#L198-L199


Not a big deal if it's not supported but more clearly a test key if we can.

**Review by ajsutton — APPROVED** [review]
LGTM. Suggest we merge the devnet-fault-proofs fix first since that's a required job which is blocking this merging.  Then we can update the GitHub config once to add devnet-plasma as a required job prior to merging.

File: .circleci/config.yml (hunk 1 above)
Comment (trianglesphere) [reply .circleci/config.yml]: PR approved. Will rebase once it's merged.

File: .circleci/config.yml (hunk 2 above)
Comment (trianglesphere) [reply .circleci/config.yml]: Yea, will have to update the other side. We also need to update loading in the files, but that is easy

File: ops-bedrock/docker-compose.yml (hunk 3 above)
Comment (trianglesphere) [reply ops-bedrock/docker-compose.yml:227]: Unfortunately not supported - sentinel isn't our piece of software

**Pull request merged**